ParProcCo/msmapper_aggregator.py

```diff
@@ -63,44 +67,48 @@ def _renormalise(self, output_data_files: List[Path]) -> None:
     def _initialise_arrays(self) -> None:
         self.axes_mins = [np.inf] * self.data_dimensions
         self.axes_maxs = [np.NINF] * self.data_dimensions
-        self._axes_starts_stops = [[] for _ in range(self.data_dimensions)]
         self.all_slices = []
 
+        data_group_name = "/".join([self.nxentry_name, self.nxdata_name])
         for data_file in self.output_data_files:
             with h5py.File(data_file, "r") as f:
-                data_group_name = "/".join([self.nxentry_name, self.nxdata_name])
-                nx_data_group = f[data_group_name]
-                volumes_array = np.array(nx_data_group[self.signal_name])
-                axes = [np.array(nx_data_group[axis_name]) for axis_name in self.axes_names]
-
-            axes_lengths = volumes_array.shape
-            slices = []
-
+                axes = [np.array(f[data_group_name][axis_name]) for axis_name in self.axes_names]
             for j, axis in enumerate(axes):
                 self.axes_mins[j] = min([min(axis), self.axes_mins[j]])
                 self.axes_maxs[j] = max([max(axis), self.axes_maxs[j]])
-                self._axes_starts_stops[j].append([axis[0], axis[-1]])
 
-                start = int((axis[0] - self.axes_mins[j]) / self.axes_spacing[j])
-                stop = axes_lengths[j] - start
+        for data_file in self.output_data_files:
+            with h5py.File(data_file, "r") as f:
+                volumes_array = np.array(f[data_group_name][self.signal_name])
+                axes = [np.array(f[data_group_name][axis_name]) for axis_name in self.axes_names]
+
+            axes_lengths = tuple(len(axis) for axis in axes)
+            assert axes_lengths == volumes_array.shape, "axes_lengths must equal volumes_array.shape"
+            slices = []
+            for j, axis in enumerate(axes):
+                start = int(round((axis[0] - self.axes_mins[j]) / self.axes_spacing[j]))
+                stop = axes_lengths[j] + start
                 slices.append(slice(start, stop))
 
-            assert len(slices) == self.data_dimensions
+            assert len(slices) == self.data_dimensions, "number of slices must equal self.data_dimensions"
```

Comment:
Effectively, this is a check of len(axes) == len(volume_shape) == self.data_dimensions